nmap_test.go

```diff
@@ -271,10 +271,10 @@ func TestRunWithProgress(t *testing.T) {
 				WithCustomArguments("tests/xml/scan_base.xml"),
 			},
 
```

Comment:
Could you please add a unit test for the new method? 🙏

Reply:
I will do so :)

Reply:
I have written a unit test for this method but sadly I can't test all the errors until the code is refactored.